Bio/SeqIO/GfaIO.py

```diff
@@ -0,0 +1,177 @@
+        elif parts[1] == "A" and re.fullmatch(r"[!-~]", parts[2]) is None:
+            warnings.warn(
+                f"Tag has incorrect type. Expected printable character, got {parts[2]}.",
+                BiopythonWarning,
+            )
+        elif parts[1] == "i" and re.fullmatch(r"[-+]?[0-9]+", parts[2]) is None:
+            warnings.warn(
+                f"Tag has incorrect type. Expected signed integer, got {parts[2]}.",
+                BiopythonWarning,
+            )
+        elif (
+            parts[1] == "f"
+            and re.fullmatch(r"[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?", parts[2])
+            is None
+        ):
+            warnings.warn(
+                f"Tag has incorrect type. Expected float, got {parts[2]}.",
+                BiopythonWarning,
+            )
+        elif parts[1] == "Z" and re.fullmatch(r"[ !-~]+", parts[2]) is None:
+            warnings.warn(
+                f"Tag has incorrect type. Expected printable string, got {parts[2]}.",
+                BiopythonWarning,
+            )
+        elif parts[1] == "J" and re.fullmatch(r"[ !-~]+", parts[2]) is None:
+            warnings.warn(
+                f"Tag has incorrect type. Expected JSON excluding new-line and tab characters, got {parts[2]}.",
+                BiopythonWarning,
+            )
+        elif parts[1] == "H" and re.fullmatch(r"[0-9A-F]+", parts[2]) is None:
+            warnings.warn(
+                f"Tag has incorrect type. Expected byte array in hex format, got {parts[2]}.",
+                BiopythonWarning,
+            )
+        elif (
+            parts[1] == "B"
+            and re.fullmatch(
+                r"[cCsSiIf](,[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?)+", parts[2]
+            )
+            is None
+        ):
+            warnings.warn(
+                f"Tag has incorrect type. Expected array of integers or floats, got {parts[2]}.",
+                BiopythonWarning,
+            )
+    return annotations
+
+
+def Gfa1Iterator(source):
+    """Parser for GFA 1.x files.
+
+    Documentation: https://gfa-spec.github.io/GFA-spec/GFA1.html
+    """
+    with as_handle(source) as handle:
+        for line in handle:
+            if line == "\n":
+                warnings.warn("GFA data has a blank line.", BiopythonWarning)
+                continue
+
+            fields = line.strip("\n").split("\t")
+            if fields[0] != "S":
+                continue
+            if len(fields) < 3:
+                raise ValueError(
+                    f"Segment line must have name and sequence fields: {line}."
+                )
+
+            if fields[2] == "*":
+                seq = Seq(None, length=0)
+            else:
+                seq = Seq(fields[2])
+
+            tags = fields[3:]
+            _check_tags(seq, tags)
+            annotations = _tags_to_annotations(tags)
+
+            yield SeqRecord(seq, name=fields[1], annotations=annotations)
```

Comment:
In general the ``.id`` is the primary identifier used in ``SeqIO``.

Leave ``.name`` as the default, or use the same value if you prefer.